cmd/thor/utils.go

```diff
@@ -534,6 +538,35 @@ func startAPIServer(ctx *cli.Context, handler http.Handler, genesisID thor.Bytes
 	}, nil
 }
 
+func startTelemetryServer(addr, allowedOrigins string) (string, func(), error) {
+	origins := strings.Split(strings.TrimSpace(allowedOrigins), ",")
+	for i, o := range origins {
+		origins[i] = strings.ToLower(strings.TrimSpace(o))
+	}
+
+	listener, err := net.Listen("tcp", addr)
+	if err != nil {
+		return "", nil, errors.Wrapf(err, "listen Telemetry API addr [%v]", addr)
+	}
+
+	router := mux.NewRouter()
+	router.PathPrefix("/metrics").Handler(telemetry.HTTPHandler())
+	handler := handlers.CompressHandler(router)
+
+	// setup cors
+	handler = handlers.CORS(handlers.AllowedOrigins(origins))(handler)
```

Comment:
I think `/metrics` does not require CORS to be set, right?

Reply:
Yeah that's correct. It would only be required if there was a page served that would initiate requests to the server.